lib/xdp_loader.h

```diff
@@ -0,0 +1,31 @@
+#ifndef __XDP_LOADER_H_
+#define __XDP_LOADER_H_
+
+using namespace std;
+
+class XdpLoader {
+    public:
+        XdpLoader(): handle(nullptr) { }
+        ~XdpLoader();
+        int LoadFrmFile(const string filepath, const int ifindex);
+        int LoadFrmBuffer(const string_view buffer, const char *name);
+    private:
+        void *handle;
+};
+
+using XdpHandle = unique_ptr<XdpLoader>;
```

Comment:
XdpLoader is not providing any value beyond ~XdpLoader(). You should use std::unique_ptr<void*, HandleDtor> as described in Carlo's previous comment.

Reply:
My intention was to enhance it later with say tail calls and beyond.
What do you think? We can also discuss F2F and decide this first.

Reply:
I think we should focus on loading XDP programs for now. As you say, we'll eventually be loading an ensemble of tail call programs, however each of the individual programs still needs to be managed so I think we should solve the XDP program lifetime in isolation.

Reply:
As discussed over email, unique ptr to void* is not viable. Retaining the class.